### ISSUE TITLE
Finetuning: Adaptive training strategy

### ISSUE THREAD

**Issue by onurgu** [boun-tabi-LMG/turkish-lm-tuner#6]
In the initial phase of fine-tuning, we've standardized the learning rate adjustment by using the AdaFactor scheduler for all tasks. This strategy aims to avoid testing various learning rates and schedulers at this stage. The AdaFactor scheduler's implementation and configuration are controlled by the **`adafactor_scheduler`** parameter, as shown in the configuration file: [AdaFactor Scheduler in default.yaml](https://github.com/boun-llm/t5-tuner/blob/c4d4e9cfc416e871bb74bcbc1b243a0fd70f0a49/conf/default.yaml#L4C1-L5C1)

Moreover, the fine-tuning process will span 10 epochs and include an early stopping mechanism. We'll monitor the model’s performance on the validation set during training, and the model checkpoint showing the best validation performance will be saved.
